rustler_codegen/src/ex_struct.rs

```diff
@@ -65,9 +65,13 @@ pub fn transcoder_decorator(ast: &syn::DeriveInput, add_exception: bool) -> Toke
 }
 
 fn gen_decoder(ctx: &Context, fields: &[&Field], atoms_module_name: &Ident) -> TokenStream {
-    let struct_type = &ctx.ident_with_lifetime;
     let struct_name = ctx.ident;
     let struct_name_str = struct_name.to_string();
+    let lifetimes = &ctx.lifetimes;
+    // FIXME maybe put in context
+    let mut rustler_decoder_lifetimes = vec![];
+    rustler_decoder_lifetimes.resize(lifetimes.len(), quote! { 'a });
+    let struct_type = quote! { #struct_name < #(#rustler_decoder_lifetimes),* > };
```

Comment:
This is an ugly hack. Basically, every lifetime in the type is set to `<'a, 'a, 'a,...>`, which is the lifetime used further below in the decoder.